### PR TITLE
Add OS-specific BAZELISK_HOME

### PR DESCRIPTION
**Pull request by mzapotoczny**
`BAZELISK_HOME` should be set as absolute path, but all supported OSes use different folder structure. This might be a problem if we want to globally set `BAZELISK_HOME` for all developers.

Therefore we want to propose a change to add support for OS-specific `BAZELISK_HOME`, for example `BAZELISK_HOME_WINDOWS`.

### CODE REVIEW

**Comment by mzapotoczny** [comment]
@fweikert could you take a look? Thanks in advance :)

**Comment by McAmun** [comment]
@katre can you maybe help here ?

File: README.md [hunk 1]
```diff
@@ -195,6 +198,8 @@ Configuration variables are evaluated with precedence order. The preferred value
 * Variables defined in the workspace root `.bazeliskrc`
 * Variables defined in the user home `.bazeliskrc`
 
+Additionally, the bazelisk home directory is also evaluated in precedence order. The preffered value is OS-specific e.g. `BAZELISK_HOME_LINUX`, then we fall back to `BAZELISK_HOME`.
```
Comment (fweikert) [inline README.md:201]: nit: preferred

Comment (fweikert) [inline README.md:201]: Bazelisk

Comment (mzapotoczny) [reply README.md]: thanks for catching it, fixed

Comment (mzapotoczny) [reply README.md]: fixed

**Comment by McAmun** [comment]
@fweikert any update on this ?

**Comment by McAmun** [comment]
@fweikert @albertocavalcante - Can we get this merged please ? We would like to start using your official releases.

File: README.md (hunk 1 above)
Comment (fweikert) [reply README.md]: Nit: preferred

**Review by fweikert — APPROVED** [review]

**Comment by fweikert** [comment]
@mzapotoczny thanks for this change, I took the liberty of resolving the merge conflict.

**Pull request merged**